draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -274,70 +291,176 @@ The array is created as follows:
+variations in the result of the JSON encoding apply.
+
+For example, a Disclosure for the second element of the `nationalities` array in the following claim set:
+
+```json
+{
+  "nationalities": ["DE", "FR"]
+}
+```
+
+could be created as follows:
+
+```json
+["lklxF5jMYlGTPUovMNIvCA", "FR"]
+```
+
+The resulting Disclosure would be: `WyJsa2x4RjVqTVlsR1RQVW92TU5JdkNBIiwgIkZSIl0`
+
+## Hashing Disclosures {#hashing_disclosures}
 
 For embedding the Disclosures in the SD-JWT, the Disclosures are hashed using the hash algorithm specified in the `_sd_alg` claim described in (#hash_function_claim). The resulting digest is then included in the SD-JWT instead of the original claim value, as described next.
 
 The digest MUST be taken over the US-ASCII bytes of the base64url-encoded Disclosure. This follows the convention in JWS [@!RFC7515] and JWE [@RFC7516]. The bytes of the digest MUST then be base64url-encoded.
 
 It is important to note that:
 
- * The input to the hash function is the base64url-encoded Disclosure, not the bytes encoded by the base64url string.
- * The bytes of the output of the hash function are base64url-encoded, and are not the bytes making up the (often used) hex representation of the bytes of the digest.
+ * The input to the hash function MUST be the base64url-encoded Disclosure, not the bytes encoded by the base64url string.
+ * The bytes of the output of the hash function MUST be base64url-encoded, and are not the bytes making up the (often used) hex representation of the bytes of the digest.
 
-For example, the
-SHA-256 digest of the Disclosure `WyI2cU1RdlJMNWhhaiIsICJmYW1pbHlfbmFtZSIsICJNw7ZiaXVzIl0` would be
+For example, the SHA-256 digest of the Disclosure
+`WyI2cU1RdlJMNWhhaiIsICJmYW1pbHlfbmFtZSIsICJNw7ZiaXVzIl0` would be
 `uutlBuYeMDyjLLTpf6Jxi7yNkEF35jdyWMn9U7b_RYY`.
 
-#### Decoy Digests {#decoy_digests}
+The SHA-256 digest of the Disclosure
+`WyJsa2x4RjVqTVlsR1RQVW92TU5JdkNBIiwgIkZSIl0` would be
+`w0I8EKcdCtUPkGCNUrfwVp2xEgNjtoIDlOxc9-PlOhs`.
 
-An Issuer MAY add additional digests to the SD-JWT that are not associated with any claim.  The purpose of such "decoy" digests is to make it more difficult for an attacker to see the original number of claims contained in the SD-JWT. It is RECOMMENDED to create the decoy digests by hashing over a cryptographically secure random number. The bytes of the digest MUST then be base64url-encoded as above. The same digest function as for the Disclosures MUST be used.
+## Embedding Disclosure Digests in SD-JWTs {#embedding_disclosure_digests}
 
-For decoy digests, no Disclosure is sent to the Holder, i.e., the Holder will see digests that do not correspond to any Disclosure. See (#decoy_digests_privacy) for additional privacy considerations.
+Digests of disclosures are embedded into the SD-JWT instead of the original
+claims. The precise way of embedding the digests depends on the type of the
+claim, as described in the following sections.
```

Comment:
```suggestion
For selectively disclosable claims, the digests of the Disclosures are embedded into the SD-JWT instead of the claims themselves. The precise way of embedding depends on whether a claim is an object property (key-value pair) or an array element.

 * For a claim that is an object property, the Issuer embeds a Disclosure as described in (#embedding_object_properties).
* For a claim that is an array element, the Issuer creates a Disclosure as described in (#embedding_array_elements).
```